### PR TITLE
Add ui segmented control

### PR DESCRIPTION
**Pull request by nfarah86**
@gwalker1228 please review!  Thanks!

### CODE REVIEW

**Comment by nfarah86** [comment]
@mplewis please review :)

File: CardMatchingGameWithBean/Model/CardMatchingGame.m [hunk 1]
```diff
@@ -45,48 +49,93 @@ -(instancetype) initWithCardCount:(NSUInteger)count usingDeck:(Deck *)deck
...
+        [self scoreGame:@"cards don't match"];
+        [self.chosenCards removeObject:cardToBeUnMatched];
+        NSLog(@"CARDS AFTER BEING REMOVED %ld", [self.chosenCards count]);
+        cardToBeUnMatched.chosen = NO;
+    }
+}
 
+- (NSInteger)_differentiateTwoAndThreeCardGames:(NSInteger)segmentControlIndex
+{
+    return [self.chosenCards count] == segmentControlIndex + 2 ? [self.chosenCards count]: 0;
```
Comment (nfarah86) [inline CardMatchingGameWithBean/Model/CardMatchingGame.m:144]: In order for me to set the segmentControlIndex properly, I'll have to add the cards in the array in the ViewController, and put the matching logic there (how cards are being flipped as they pick it.).  As the user are picking cards, the cards are flipping over.  If I add the cards in an array in the ViewController, then handle the flipping logic in this current file, the cards won't flip when they pick it. It'll flip afterwards.

File: CardMatchingGameWithBean/Model/CardMatchingGame.m [hunk 2]
```diff
@@ -45,48 +49,92 @@ -(instancetype) initWithCardCount:(NSUInteger)count usingDeck:(Deck *)deck
...
+            card.chosen = YES;
+            NSInteger cardsInArray =
+                [self _differentiateTwoAndThreeCardGames:
+                          segmentedControlIndex];
+            if (cardsInArray) {
+                self.matchScore = [card match:self.chosenCards];
+                if (self.matchScore) {
+                    [self matchCards:self.chosenCards];
+                } else if (self.matchScore == 0) {
+                    [self unmatchCards:self.chosenCards];
```
Comment (mplewis) [inline CardMatchingGameWithBean/Model/CardMatchingGame.m:97]: `0` will automatically evaluate to `NO` in an if statement, so you don't have to use else if == 0, you can just use else

**Comment by nfarah86** [comment]
@mplewis I added a label that describes which cards are being flipped; and implemented a delegate protocol.  I used a delegate protocol to send that cards that are picked by the user to the view controller.  Then I write to the label which cards the user picked, and whether they have a match or not.

**Pull request merged**